server/src/main/java/org/elasticsearch/gateway/GatewayMetaState.java

```diff
@@ -252,6 +257,14 @@ private static boolean applyPluginUpgraders(ImmutableOpenMap<String, IndexTempla
         return false;
     }
 
+    @Override
+    public void close() throws IOException {
+        final PersistedState persistedState = this.persistedState.get();
```

Comment:
NIT: could just use `IOUtils.close(this.persistedState.get()));` here for the `null` check

Reply:
Thanks, see 0639b33be68454408d084d05afcfc41a90acb840